### PR TITLE
wip

### PR DESCRIPTION
**Pull request by dsh**

### CODE REVIEW

File: src/SellerLabs/Slapp/Auth/SlappGuard.php [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+        $id = $this->getData(self::USER_ID);
+
+        $user = null;
+        if (! is_null($id)) {
+            $user = $this->provider->findBySlappUserId($id);
+        }
+        return $this->user = $user;
+    }
+
```
Comment (5outh) [inline src/SellerLabs/Slapp/Auth/SlappGuard.php]: Normally I wouldn't mention it, but there's a lot of excess whitespace here

File: src/SellerLabs/Slapp/Auth/SlappGuard.php [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+     * @return mixed|null
+     */
+    protected function getData($name)
+    {
+        if (!$this->dataLoaded) {
+            try {
+                $c = array_get($_COOKIE, $this->getName());
+                $this->data = json_decode($this->encrypter->decrypt($c), true);
+            } catch (Exception $e) {
+                // ignore
```
Comment (5outh) [inline src/SellerLabs/Slapp/Auth/SlappGuard.php]: Why?

File: src/SellerLabs/Slapp/Auth/SlappGuard.php [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+     *
+     * @param SlappUserInterface $user
+     */
+    public function login(SlappUserInterface $user)
+    {
+        // If my sub-domain and the SLAPP sub-domain share the same top level
+        // domain, then create the cookie to be shared across the entire
+        // top level domain. E.g. scope.sellerlabs.com and accounts.sellerlabs.com
+        // will both put the cookie in .sellerlabs.com.
+        $top = function($domain) {
```
Comment (5outh) [inline src/SellerLabs/Slapp/Auth/SlappGuard.php]: Looks like domain construction could be extracted to a function or class

**Comment by 5outh** [comment]
Looks good to me, thanks!

File: src/SellerLabs/Slapp/Auth/SlappGuard.php (hunk 2 above)
Comment (dsh) [inline src/SellerLabs/Slapp/Auth/SlappGuard.php]: I'll add a comment in the code.

If the decryption or json_decode throws an error, that means the cookie is corrupted some how. Effectively, we'll just ignore it and pretend the cookie doesn't even exist.

If the cookie is corrupt, something went wrong, and we can't really recover.

**Pull request merged**